ipsource.go

```diff
@@ -38,7 +38,13 @@ func init() {
 
 // IPSource is a type that can get IP addresses.
 type IPSource interface {
-	GetIPs(context.Context, IPVersions) ([]net.IP, error)
+	GetIPs(context.Context, IPSettings) ([]net.IP, error)
+}
+
+// Configuration for enabled IP versions and IP range filtering.
+type IPSettings struct {
+	*IPRanges
```

Comment:
Why is this a pointer?

Reply:
It was a while ago so I'm not 100% sure I remember correctly, but I think it was because `*IPRanges` had to be a pointer in the `App` struct to get `omitempty` to work, and so this made it easier to construct `IPSettings{a.IPRanges, a.Versions}` as a one-liner.

But now it seems we can use `omitzero` and so the bad incentive is removed.